src/Executor/ScopedContext.php

```diff
@@ -0,0 +1,12 @@
+<?php declare(strict_types=1);
+
+namespace GraphQL\Executor;
+
+/**
+ * This interface can be used to indicate that your context should be scoped.
+ *
+ * A scoped context will only delegate information to children.
+ */
+interface ScopedContext
```

Comment:
Should we maybe forbid the use of the non-scoped one? Since it doens't really make sense to use it in 99% of cases I think.

Reply:
This PR targets `webonyx/graphql-php`, we'll only use scoped context in our implementation.